backend/capellacollab/core/authentication/provider/azure/routes.py

```diff
@@ -22,15 +23,21 @@
 from capellacollab.users.models import Role
 
 router = fastapi.APIRouter()
-cfg = config["authentication"]["azure"]
+cfg = (
+    config.authentication.azure
+    if isinstance(
+        config.authentication, config_models.AzureAuthenticationConfig
+    )
+    else None
+)
 
 
 @lru_cache
 def ad_session():
     return ConfidentialClientApplication(
-        cfg["client"]["id"],
-        client_credential=cfg["client"]["secret"],
-        authority=cfg["authorizationEndpoint"],
+        cfg.client.id if cfg else None,
+        client_credential=cfg.client.secret if cfg else None,
+        authority=cfg.authorizationEndpoint if cfg else None,
```

Comment:
The Azure | OAuth config union raises attribute errors in the few places we were already expecting Azure or OAuth configuration. This is obviously an inelegant solution, would it be better to throw an error before the ConfidentialClientApplication return if we're using the wrong config here?

Reply:
```
def ad_session():
     if cfg is None:
         raise ValueError("Configuration for Azure is not set.")
```
Changed it to use a basic exception, I don't know how thorough you want it to be since I'm not sure how someone would get to the azure route without the proper config anyway